src/Stratis.Bitcoin.Features.Interop/EthereumClient/MultisigWallet.cs

```diff
@@ -0,0 +1,732 @@
+    public class ExecuteTransactionFunction : FunctionMessage
+    {
+        [Parameter("uint256", "transactionId", 1)]
+        public BigInteger TransactionId { get; set; }
+    }
+
+    [Function("getConfirmationCount", "uint256")]
+    public class GetConfirmationCountFunction : FunctionMessage
+    {
+        [Parameter("uint256", "transactionId", 1)]
+        public BigInteger TransactionId { get; set; }
+    }
+
+    public class MultisigWallet
+    {
+        public static async Task<string> DeployContract(Web3 web3, string[] owners, uint required)
+        {
+            var deploymentMessage = new MultisigWalletDeployment()
+            {
+                Owners = owners,
+                Required = required
+            };
+
+            IContractDeploymentTransactionHandler<MultisigWalletDeployment> deploymentHandler = web3.Eth.GetContractDeploymentHandler<MultisigWalletDeployment>();
+            TransactionReceipt transactionReceiptDeployment = await deploymentHandler.SendRequestAndWaitForReceiptAsync(deploymentMessage);
+            string contractAddress = transactionReceiptDeployment.ContractAddress;
+
+            return contractAddress;
+        }
+
+        public static async Task<List<string>> GetOwners(Web3 web3, string contractAddress)
+        {
+            var getOwnersFunctionMessage = new GetOwnersFunction()
+            {
+            };
+
+            IContractQueryHandler<GetOwnersFunction> ownerHandler = web3.Eth.GetContractQueryHandler<GetOwnersFunction>();
+            List<string> owners = await ownerHandler.QueryAsync<List<string>>(contractAddress, getOwnersFunctionMessage);
+
+            return owners;
+        }
+
+        public static async Task<BigInteger> SubmitTransaction(Web3 web3, string contractAddress, string destination, BigInteger value, string data, BigInteger gas, BigInteger gasPrice)
+        {
+            IContractTransactionHandler<SubmitTransactionFunction> submitHandler = web3.Eth.GetContractTransactionHandler<SubmitTransactionFunction>();
+
+            var submitTransactionFunctionMessage = new SubmitTransactionFunction()
+            {
+                Destination = destination,
+                Value = value,
+                Data = ConvertHexStringToByteArray(data),
+                Gas = gas,
+                GasPrice = Web3.Convert.ToWei(gasPrice, UnitConversion.EthUnit.Gwei)
+            };
+
+            TransactionReceipt submitTransactionReceipt = await submitHandler.SendRequestAndWaitForReceiptAsync(contractAddress, submitTransactionFunctionMessage);
+            EventLog<SubmissionEventDTO> submission = submitTransactionReceipt.DecodeAllEvents<SubmissionEventDTO>().FirstOrDefault();
+
+            // Use -1 as an error indicator.
+            if (submission == null)
+                return BigInteger.MinusOne;
+
+            return submission.Event.TransactionId;
+        }
+
+        private static byte[] ConvertHexStringToByteArray(string hexString)
```

Comment:
I'm pretty sure we have implementation of HEX => Byte array somewhere else in the codebase already